### ISSUE TITLE
Unable to build jekyll site locally

### ISSUE THREAD

**Issue by DirtyF** [jekyll/jekyll#5755]
- [x] I am on (or have tested on) ***macOS*** 10+
- [x] I was trying to build.

## My Reproduction Steps

```shell
$ ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

$ bundle exec jekyll -v
jekyll 3.3.1

$ bundle exec rake site:preview
Running Jekyll...
Configuration file: /Users/frank/code/jekyll/jekyll/docs/_config.yml
            Source: /Users/frank/code/jekyll/jekyll/docs
       Destination: /Users/frank/code/jekyll/jekyll/docs/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
rake aborted!
NoMethodError: undefined method `gsub' for nil:NilClass
/Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:174:in `block in generate'
/Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:172:in `each'
/Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:172:in `generate'
/Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:68:in `process'
/Users/frank/code/jekyll/jekyll/lib/jekyll/command.rb:26:in `process_site'
/Users/frank/code/jekyll/jekyll/lib/jekyll/commands/build.rb:63:in `build'
/Users/frank/code/jekyll/jekyll/lib/jekyll/commands/build.rb:34:in `process'
/Users/frank/code/jekyll/jekyll/rake/site.rake:31:in `block (2 levels) in <top (required)>'
/Users/frank/.rbenv/versions/2.3.1/bin/bundle:22:in `load'
/Users/frank/.rbenv/versions/2.3.1/bin/bundle:22:in `<main>'
Tasks: TOP => site:preview
(See full trace by running task with --trace)

$ script/proof
==> Installing...
==> Building...
Configuration file: docs/_config.yml
            Source: docs
       Destination: _site
 Incremental build: disabled. Enable with --incremental
      Generating...
bundler: failed to load command: jekyll (/Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/bin/jekyll)
NoMethodError: undefined method `gsub' for nil:NilClass
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/html-pipeline-2.4.2/lib/html/pipeline/emoji_filter.rb:46:in `emoji_image_filter'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/jemoji-0.5.1/lib/jemoji.rb:34:in `emojify'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/jemoji-0.5.1/lib/jemoji.rb:29:in `block in generate'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/jemoji-0.5.1/lib/jemoji.rb:29:in `each'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/jemoji-0.5.1/lib/jemoji.rb:29:in `generate'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:174:in `block in generate'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:172:in `each'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:172:in `generate'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/site.rb:68:in `process'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/command.rb:26:in `process_site'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/commands/build.rb:63:in `build'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/commands/build.rb:34:in `process'
  /Users/frank/code/jekyll/jekyll/lib/jekyll/commands/build.rb:16:in `block (2 levels) in init_with_program'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
  /Users/frank/code/jekyll/jekyll/exe/jekyll:13:in `<top (required)>'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/bin/jekyll:22:in `load'
  /Users/frank/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/bin/jekyll:22:in `<top (required)>'
```

/cc @jekyll/build @jekyll/core

**Comment by Crunch09** [comment]
Can confirm this. It seems that the `page.content` of [github.html](https://github.com/jekyll/jekyll/blob/master/docs/redirects/github.html) is `null` where before it was a String.

**Comment by pathawks** [comment]
Do we not have a CI test for this‽ :feelsgood:

**Comment by ashmaroli** [comment]
**`not-reproduced`** on **`Windows`**  
I was able to `build` successfully and was `served` as expected by `bundle exec rake site:preview`

**Comment by Crunch09** [comment]
This issue is caused by https://github.com/jekyll/jekyll-redirect-from/commit/cb30dcf59dcd9a71d4fcdc63ee62cdcb25e02a7f . I'll try to come up with a fix. cc: @benbalter 

@ashmaroli You probably used `jekyll-redirect-from` `0.11.0`. Can you confirm this issue is also occurring on windows if you use `0.12.0` ?

**Comment by ashmaroli** [comment]
@Crunch09 I tried again with `jekyll-redirect-from` `0.12.0`... 
This issue is still **`not-reproduced`** on **`Windows`**

--
Note: The only gem not updated is `pygments.rb`. Its still at `0.6.3` for Windows (symlinks issue with `v1.1.1`)

**Comment by benbalter** [comment]
Released [Jekyll Redirect From v0.12.1](https://github.com/jekyll/jekyll-redirect-from/releases/tag/v0.12.1) which includes this fix (as well as a fix for Ruby 2.0.x and 2.1.x).

Sorry for the trouble (think this one was my fault :smile:).

**Comment by benbalter** [comment]
Reading the backtrace more closely, I believe this should only affect sites that have other plugins that fire *after* Jekyll Redirect From that expect `Page#output` to be a string, and not nil (in this case Jemoji) during the generate step. While this should fix the immediate issue, it may be wise to also guard against `output` being nil in Jemoji as well.

**Comment by Crunch09** [comment]
@benbalter
> [...] it may be wise to also guard against output being nil in Jemoji as well.

Looking at jemoji, i just saw we already [have this guard](https://github.com/jekyll/jemoji/blame/11ca77eecbe2a0fd6bbeb91365ffca35eaf3a42f/lib/jemoji.rb#L13) in `jemoji`. It is just not included in `0.5.1` which `jekyll` used until [earlier today](https://github.com/jekyll/jekyll/commit/c39414a17b43399243190a31df7dfe24eeb3d945).
